router/io-spec-abstract-transitions.gobra

```diff
@@ -50,6 +50,17 @@ pure func AbsValidateIngressIDConstraint(pkt io.IO_pkt2, ingressID option[io.IO_
 			ingressID == (currseg.ConsDir ? currseg.Future[0].InIF2 : currseg.Future[0].EgIF2)
 }
 
+ghost
+opaque
+requires pkt.RightSeg != none[io.IO_seg2]
```

Comment:
if we require this to not be `none`, why don't we pass an `io.IO_ifs` instead of an `option`?